CatAnalyzer/plugins/ttbbLepJetsAnalyzer.cc

```diff
@@ -732,10 +778,46 @@ void ttbbLepJetsAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetu
     // DR 
     b_DRAddJets = genttbarConeCat->begin()->dRaddJets();
 
+    // adding b jet four-momentum having minimum deltaR
+
+    std::vector<math::XYZTLorentzVector> genbjets = genttbarConeCat->begin()->bJets();
+    if( genbjets.size() > 0 ){
+      double mindR=999;
+      int index1=-1, index2=-1;
+      for( unsigned int i=0; i < genbjets.size(); ++i ){
```

Comment:
1. The loop will run only when the number of genbjets is more than 0.
2. index1 and index2 are initialized as -1. And branch will be filled when index is not -1. Therefore analyzer does not return wrong answer(maybe).